components/ai_chat/resources/page/components/markdown_renderer/index.tsx

```diff
@@ -0,0 +1,76 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import Markdown from 'react-markdown'
+
+import styles from './style.module.scss'
+import CaretSVG from '../svg/caret'
+
+const CodeBlock = React.lazy(async () => ({
+  default: (await import('../code_block')).default.Block
+}))
+const CodeInline = React.lazy(async () => ({
+  default: (await import('../code_block')).default.Inline
+}))
+
+interface MarkdownRendererProps {
+  text: string
+  shouldShowTextCursor: boolean
+}
+
+export default function MarkdownRenderer(mainProps: MarkdownRendererProps) {
+  const [lastLine, setLastLine] = React.useState(1)
+
+  const plugin = React.useCallback(() => {
+    const transformer = (ast: any) => {
+      setLastLine(ast.position.end.line)
+    }
+
+    return transformer
+  }, [mainProps.text])
+
+  return (
+    <div className={styles.markdownContainer}>
+      <Markdown
+        rehypePlugins={[plugin]}
+        disallowedElements={['a']}
```

Comment:
As mentioned on the review: are `img` elements `![alt](url)` intentionally allowed? If not, we should disallow them.